cpp/test/sg/rproj_test.cu

```diff
@@ -76,10 +74,9 @@ class RPROJTest : public ::testing::Test {
     };
 
     cudaStream_t stream = h.get_stream();
-    auto alloc          = h.get_device_allocator();
-    random_matrix1      = new rand_mat<T>(alloc, stream);
+    random_matrix1      = new rand_mat<T>(stream);
```

Comment:
While we're changing this, maybe we can take care of the raw `new` at the same time? `unique_ptr` should work just fine to replace the `new`s that we're touching throughout this file.